llvm/lib/Target/AArch64/AArch64InstrInfo.cpp

```diff
@@ -4089,6 +4089,11 @@ AArch64InstrInfo::getLdStOffsetOp(const MachineInstr &MI) {
   return MI.getOperand(Idx);
 }
 
+const MachineOperand &
+AArch64InstrInfo::getLdStAmountOp(const MachineInstr &MI) {
+  return MI.getOperand(4);
```

Comment:
How do you know this is always going to be operand 4 without inspecting the opcode? For example, see the functions above.

Reply:
yes, it need extend with a `switch case ` when we add more supported instrunction. 
The first version I only supported the `LDRBBroX `,  and it is **operand 4**.
I only add one instrunction in the first version to make sure if it is reasonable for the optimization direction itself.
If it is adopted, I will add other memory access instructions in the subsequent patch.

Reply:
OK, but my point is that if your patch lands another engineer may start using this function believing that it is complete. However, it might do the wrong thing if called for a different opcode. I think you need to explicitly add an error if the opcode is not what you expect so that it fails sensibly.

Reply:
Thanks for your idea, fixed with commit  01029d9